### PR TITLE
json status updates

### LINKED ISSUE #9: Output progress updates regarding pcap processing
In the era when pcap processing was handled via `zqd`, progress updates were made available so the Brim app could show the user how long import was taking and show partial results during import. We expect we can provide equivalent functionality in the Brimcap era. @mccanne's suggested approach is to deliver these in the form of updates over stdout from `brimcap load`.

### PR DESCRIPTION
**Pull request by mattnibs**
If brimcap analyze or load is not writing to stdout (load never is), and if
stdout is not a terminal (or the -json flag is enabled) write json status
updates to stderr.

The json status messages are mostly the same as the status updates sent back by
the zqd pcap post endpoint; this should be an easy transition for the when the
brim desktop application starts calling brimcap directly.

Closes #9

### CODE REVIEW

File: cmd/brimcap/analyze/command.go [hunk 1]
```diff
@@ -58,19 +62,25 @@ func (c *Command) Run(args []string) (err error) {
...
+	defer pcapfile.Close()
+
+	display := analyzecli.NewDisplay(c.JSON, pcapsize)
+	zctx := resolver.NewContext()
+	analyzer := analyzer.CombinerWithContext(ctx, zctx, pcapfile, c.analyzeflags.Configs...)
 
 	// If not emitting to stdio write stats to stderr.
 	if c.out.FileName() != "" {
-		c.analyzer.RunDisplay()
+		display.Run(analyzer)
```
Comment (nwt) [inline cmd/brimcap/analyze/command.go:77]: Maybe do everything with `display` here?
```suggestion
		display := analyzecli.NewDisplay(c.JSON, pcapsize)
		display.Run(analyzer)
		defer display.Close()
```

**Review by nwt — APPROVED** [review]

File: cli/analyzecli/display.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	if status.WarningsCount > 0 {
+		fmt.Fprintf(w, "warnings=%d", status.WarningsCount)
+	}
+	io.WriteString(w, "\n")
+	return true
+}
+
+func (a *Display) jsonDisplay(w io.Writer) bool {
+	return true
+}
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply cli/analyzecli/display.go:77]: Not it's not!

File: cli/analyzecli/display.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	if a.json {
+		return json.NewEncoder(os.Stderr).Encode(MsgWarning{
+			Type:    "warning",
+			Warning: msg,
+		})
+	}
+	a.warningsMu.Lock()
+	if a.warnings == nil {
+		a.warnings = make(map[string]int)
+	}
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply cli/analyzecli/display.go:99]: Sure

File: cli/analyzecli/display.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	a.warnings[msg]++
+	atomic.AddInt32(&a.warningsCount, 1)
+	a.warningsMu.Unlock()
```
[the remark this answers is not in the document]
Comment (mattnibs) [reply cli/analyzecli/display.go:102]: Good call

File: cmd/brimcap/analyze/command.go (hunk 1 above)
Comment (mattnibs) [reply cmd/brimcap/analyze/command.go:77]: That'll work!

**Pull request merged**